src/main/java/io/cdap/plugin/aws/s3/sink/S3BatchSink.java

```diff
@@ -60,12 +62,29 @@ public class S3BatchSink extends AbstractFileSink<S3BatchSink.S3BatchSinkConfig>
   private static final String ACCESS_CREDENTIALS = "Access Credentials";
 
   private final S3BatchSinkConfig config;
+  private Asset asset;
 
   public S3BatchSink(S3BatchSinkConfig config) {
     super(config);
     this.config = config;
   }
 
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    // create asset for lineage
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(config.getPath())
+      : config.getReferenceName();
+    asset = Asset.builder(referenceName)
+      .setFqn(config.getPath().replace("s3a", "s3")).build();
```

Comment:
shouldn't replace the characters like this, there can be path like `s3://bucket/s3aFolder`, use `S3Path` to replace